api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceServiceImpl.java

```diff
@@ -189,4 +190,59 @@ public Workspace updateUserRoles(Workspace workspace, Set<WorkspaceUserRole> use
     }
     return this.saveWithLastModified(workspace);
   }
+  @Override
+  public WorkspaceAccessLevel getWorkspaceAccessLevel(String workspaceNamespace, String workspaceId) {
+    String userAccess;
+    try {
+      userAccess = fireCloudService.getWorkspace(
+          workspaceNamespace, workspaceId).getAccessLevel();
+    } catch (org.pmiops.workbench.firecloud.ApiException e) {
+      if (e.getCode() == 404) {
+        throw new NotFoundException(String.format("Workspace %s/%s not found",
+            workspaceNamespace, workspaceId));
+      } else {
+        throw new ServerErrorException(e.getResponseBody());
+      }
+    }
+    return WorkspaceAccessLevel.fromValue(userAccess);
+  }
+
+  @Override
+  public WorkspaceAccessLevel enforceWorkspaceAccessLevel(String workspaceNamespace,
+      String workspaceId, WorkspaceAccessLevel requiredAccess) {
+    WorkspaceAccessLevel access;
+    try {
+      access = getWorkspaceAccessLevel(workspaceNamespace, workspaceId);
+    } catch (Exception e) {
+      throw e;
+    }
+
+    if (workspaceAccessLevelCompare(requiredAccess, access) == -1) {
```

Comment:
As long as the enum values are in order of least privileged to most, I think you can just do requiredAccess.compareTo(access) <= 0 and get rid of workspaceAccessLevelCompare.

Reply:
`requiredAccess.compareTo(access) < 0`, not `<=` right?

Reply:
0 means "they are the same". So no, I think it's <= 0.

Reply:
Ah, yeah. It passes when <= 0. (I was testing for failure here.)